ledger-core/virtual/integration/method_test.go

```diff
@@ -348,867 +390,540 @@ func TestVirtual_CallMethodAfterPulseChange(t *testing.T) {
+			Method_PrepareObject(ctx, server, payload.Ready, objectBGlobal)
+
+			// add mock
+			outgoingCall := execution.NewRPCBuilder(gen.UniqueGlobalRef(), objectAGlobal).CallMethod(objectBGlobal, class, "Bar", []byte{})
+			objectAExecutionMock := runnerMock.AddExecutionMock("Foo")
+			objectAExecutionMock.AddStart(
+				func(ctx execution.Context) {
+					require.Equal(t, objectAGlobal, ctx.Request.Callee)
+					logger.Debug("ExecutionStart [A.Foo]")
+				},
+				&execution.Update{
+					Type:     execution.OutgoingCall,
+					Error:    nil,
+					Outgoing: outgoingCall,
+				},
+			)
+			objectAExecutionMock.AddContinue(
+				func(result []byte) {
+					logger.Debug("ExecutionContinue [A.Foo]")
+				},
+				&execution.Update{
+					Type:   execution.Done,
+					Result: requestresult.New([]byte("finish A.Foo"), objectAGlobal),
+				},
+			)
 
-		runnerMock.AddExecutionClassify("Foo", contract.MethodIsolation{Interference: contract.CallTolerable, State: contract.CallDirty}, nil)
-	}
+			runnerMock.AddExecutionMock("Bar").AddStart(
+				func(ctx execution.Context) {
+					require.Equal(t, objectBGlobal, ctx.Request.Callee)
+					logger.Debug("ExecutionStart [B.Bar]")
+				},
+				&execution.Update{
+					Type:   execution.Done,
+					Result: requestresult.New([]byte("finish B.Bar"), objectBGlobal),
+				},
+			)
 
-	// add checks to typedChecker
-	{
-		typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
-			assert.Equal(t, objectBGlobal, request.Callee)
-			assert.Equal(t, objectAGlobal, request.Caller)
-			assert.Equal(t, []byte("123"), request.Arguments)
-			assert.Equal(t, payload.CTMethod, request.CallType)
-			assert.Equal(t, uint32(1), request.CallSequence)
-			assert.Equal(t, outgoingCallRef, request.CallReason)
-			assert.Equal(t, callFlags, request.CallFlags)
-			assert.Equal(t, server.GetPulse().PulseNumber, request.CallOutgoing.Pulse())
-			return true // resend
-		})
-		typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
-			assert.Equal(t, callFlags, res.CallFlags)
+			runnerMock.AddExecutionClassify("Foo", test.flagsA, nil)
+			runnerMock.AddExecutionClassify("Bar", test.flagsB, nil)
 
-			switch res.Callee {
-			case objectAGlobal:
-				require.Equal(t, []byte("finish A.New"), res.ReturnArguments)
-				require.Equal(t, payload.CTConstructor, res.CallType)
-				require.Equal(t, server.GlobalCaller(), res.Caller)
-				require.Equal(t, outgoingA, res.CallOutgoing)
-			case objectBGlobal:
-				require.Equal(t, []byte("finish B.Foo"), res.ReturnArguments)
-				require.Equal(t, payload.CTMethod, res.CallType)
-				require.Equal(t, objectAGlobal, res.Caller)
-				require.Equal(t, server.GetPulse().PulseNumber, res.CallOutgoing.Pulse())
+			// checks
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+			typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
+				assertCalleeRequest(t, objectAGlobal, objectBGlobal, request, test.flagsA, server)
```

Comment:
maybe it's better to name it `assertVCallRequest`

Reply:
done